test/authorizable.spec.ts

```diff
@@ -0,0 +1,316 @@
+    });
+
+    it("removes address from authorities array", async () => {
+      await subject();
+
+      const authoritiesArray = await authorizable.getAuthorizedAddresses.callAsync();
+
+      expect(authoritiesArray.length).to.eql(0);
+    });
+
+    it("emits correct AuthorizedAddressRemoved log", async () => {
+      const txHash = await subject();
+      const formattedLogs = await getFormattedLogsFromTxHash(txHash);
+      const expectedLogs = getExpectedRmvAuthorizedLog(
+        authorizedAccount,
+        caller,
+        authorizable.address,
+      );
+
+      expect(JSON.stringify(formattedLogs)).to.eql(JSON.stringify(expectedLogs));
+    });
+
+    describe("when the caller is not the owner of the contract", async () => {
+      before(async () => {
+        caller = otherAccount;
+      });
+
+      it("should revert", async () => {
+        await expectRevertError(subject());
+      });
+    });
+
+    describe("when the passed address is not authorized", async () => {
+      beforeEach(async () => {
+        passedAddress = otherAccount;
+      });
+
+      it("should revert", async () => {
+        await expectRevertError(subject());
+      });
+    });
+  });
+
+  describe("#removeAuthorizedAddressAtIndex", async () => {
+    beforeEach(async () => {
+      await deployAuthorizable();
+
+      const authAccountArray: Address[] = [authAccount1, authAccount2, authorizedAccount];
+      for (const account of authAccountArray) {
+        await addAuthorizableAddress(account);
+      }
+    });
+
+    afterEach(async () => {
+      caller = ownerAccount;
+      passedAddress = authorizedAccount;
+      index = new BigNumber(2);
+    });
+
+    let caller: Address = ownerAccount;
+    let passedAddress: Address = authorizedAccount;
+    let index: BigNumber = new BigNumber(2);
+
+    async function subject(): Promise<string> {
+      return authorizable.removeAuthorizedAddressAtIndex.sendTransactionAsync(
+        passedAddress,
+        index,
+        { from: caller },
+      );
+    }
+
+    it("removes address from authorized mapping", async () => {
+      await subject();
+
+      const storedAuthAddress = await authorizable.authorized.callAsync(
+        authorizedAccount,
+      );
+
+      expect(storedAuthAddress).to.eql(false);
+    });
+
+    it("removes address from authorities array", async () => {
+      const oldAuthoritiesArray = await authorizable.getAuthorizedAddresses.callAsync();
+
+      await subject();
+
+      const newAuthoritiesArray = await authorizable.getAuthorizedAddresses.callAsync();
+
+      oldAuthoritiesArray.splice(index.toNumber(), 1);
+      expect(newAuthoritiesArray).to.eql(oldAuthoritiesArray);
+      expect(newAuthoritiesArray.length).to.eql(oldAuthoritiesArray.length);
+    });
+
+    it("emits correct AuthorizedAddressRemoved log", async () => {
+      const txHash = await subject();
+      const formattedLogs = await getFormattedLogsFromTxHash(txHash);
```

Comment:
Let's see if we can standardize log testing. It would be nice to have a helper function that takes a transaction hash, and the three parameters to `getExpectedRmvAuthorizedLog` and does the expectation. See `tokenAssertions.ts`